src/Symfony/Component/Console/Formatter/OutputFormatterInterface.php

```diff
@@ -53,6 +53,8 @@ public function getStyle(string $name);
 
     /**
      * Formats a message according to the given styles.
+     *
+     * @return string|null
```

Comment:
I think this must be `@return string` and the null formatter should be `return $message;` instead.

https://github.com/symfony/symfony/blob/49a076a120147727ed3759c678c06d79ed922521/src/Symfony/Component/Console/Output/Output.php#L157

Reply:
Keep in mind that this is not an internal class. You are suggesting to significantly change its behavior here while I'm merely documenting the status quo.

Reply:
Hmm, indeed. I'm not sure what is the best way forward.

If we officially document this contract as returning `string|null`, we have to update all code using this function to take care of the `null` return. Honestly, returning null (void actually) in `NullOutputFormatter` seems like a mistake and bug to me.

In fact, if code is build to use the normal `OutputFormatter`, it would be broken if `NullOutputFormatter` is used in the test. I can't think of any reasonable approach to build code to only work with `NullOutputFormatter`.

Reply:
I understand. But `string|null` is what currently is returned from the implementations. Once this PR is merged, we can think about how we could narrow it down to just `string` in 6.0. By the way, I'd also like to do this for the `$message` parameter.

Reply:
Should we depreciate null in 5.4 then? Then we can keep the implied contract, which is string IMHO

Reply:
I agree that the real issue is the implementation of `NullOutputFormatter` which was written before void was introduced. Conceptually, doing nothing was the same as returning `null` back then. Let's merge this one as is and see if we can deprecate `null` in a followup PR (same for `*message`).